flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -2378,6 +2378,89 @@ bool OmpStructureChecker::CheckIntrinsicOperator(
   return false;
 }
 
+static bool isReductionAllowedForType(
+    const parser::OmpClause::Reduction &x, const DeclTypeSpec *type) {
+  assert(type && "no type for reduction symbol");
```

Comment:
Semantics uses a `CHECK()` macro.  But maybe you'd be better off passing a reference rather than a pointer.  Use `DEREF()` to check a pointer and turn it into a reference if you're confident that the pointer will never be null, even in an erroneous program.